src/_types.ts

```diff
@@ -18,4 +18,4 @@ export type AssertEqual<Type, Expected> = Array<Type> extends Array<Expected>
     : never
   : never;
 
-export type NonEmptyArray<T> = [T, ...Array<T>] | [...Array<T>, T];
```

Comment:
I had to revert the changes here, there's a difference in how typescript treats the array when it's a return value (as in groupBy) and when it's used as an input value (like I did with chunk).